subsys/testsuite/ztest/src/ztest.c

```diff
@@ -1321,15 +1326,57 @@ static int cmd_shuffle(const struct shell *sh, size_t argc, char **argv)
 
 static int cmd_run_suite(const struct shell *sh, size_t argc, char **argv)
 {
+	struct getopt_state *state;
+	int opt;
+	static struct option long_options[] = {{"repeat_iter", required_argument, 0, 'r'},
+		{0, 0, 0, 0}};
+	int opt_index = 0;
+	int val;
+	int opt_num = 0;
+	void *param = NULL;
+	int repeat_iter = 1;
+
+	while ((opt = getopt_long(argc, argv, "r:p:", long_options, &opt_index)) != -1) {
+		state = getopt_state_get();
+		switch (opt) {
+		case 'r':
+			val = atoi(state->optarg);
+			if (val < 1) {
+				shell_fprintf(sh, SHELL_ERROR,
+					"Invalid number of suite interations\n");
+				return -ENOEXEC;
+			}
+			repeat_iter = val;
+			opt_num++;
+			break;
+		case 'p':
+			param = state->optarg;
+			opt_num++;
+			break;
+		default:
+			shell_fprintf(sh, SHELL_ERROR,
+				"Invalid option or option usage: %s\n", argv[opt_index + 1]);
+			return -ENOEXEC;
+		}
+	}
 	int count = 0;
 	bool shuffle = false;
+	const char *shell_command = argv[0];
 
-	ztest_set_test_args(argv[1]);
+	if (opt_num == 1) {
+		ztest_set_test_args(argv[3]);
+	} else {
+		ztest_set_test_args(argv[1]);
+	}
```

Comment:
I'd add a short comment here.

Reply:
Added the short explanation.

Reply:
The comment now explains the 'what' and not the 'why' of the operation.